fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt

```diff
@@ -105,16 +108,14 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         lifecycleScope.launch {
-            val debugDataStore = requireContext().debugSettings
-
             // During initial development, this will only be available in Nightly or Debug builds.
             requirePreference<SwitchPreference>(R.string.pref_key_enable_debug_drawer).apply {
                 isVisible = Config.channel.isNightlyOrDebug
-                isChecked = debugDataStore.debugDrawerEnabled.first()
+                isChecked = debugSettingsCache.debugDrawerEnabledFlow.first()
                 onPreferenceChangeListener =
                     Preference.OnPreferenceChangeListener { _, newValue ->
                         lifecycleScope.launch {
```

Comment:
With this [comment](https://github.com/mozilla-mobile/firefox-android/pull/4829/files#r1426849007), it will also prevent launching a new coroutine here on every pref update, since that is now happening in the new abstraction